### DISCUSSION TITLE
Best way to run subset of packages

### DISCUSSION THREAD

**Discussion by souporserious** [vercel/turborepo#176] [Help] [answered]
First, thank you for open sourcing this wonderful project! I have it set up on my more simple projects and things are working great! I've come into a more complex use case and am having trouble understanding the best way to set things up. I have a monorepo where we need to run a subset of packages and their dependencies, but not sure if I should be trying to use `scope` for that or possibly wait for the new filter option in #81 to land?

The project is set up similar to this:

packages/
├─ cli
├─ docs
├─ design-system
├─ figma-plugin
├─ icons

Where I want to do things like run `site dev` or `figma-plugin dev` on their own which each rely on `design-system` and `design-system` depends on `icons`. Right now I can easily run everything all at once, which seems ok, but wondering if it's best to use separate tasks (`yarn dev:docs`, etc) where it's only running scripts for a subset of packages.

**Comment by jaredpalmer** [comment, marked as answer]
Definitely feel the pain. 

Yes you can separate into `yarn dev:docs` and `yarn dev:figma` etc  and have those use `--scope` (also recall that you can pass multiple `--scope` flags).

We want to land `--filter`  #81 and also #144 which i think will be massive quality of life changes.

**Reply by souporserious** [reply]
Awesome, sounds good! For now, with using `scope`, I'm not sure if I'm misunderstanding how to write the pattern, but how can I scope multiple sub-packages?

The following works fine for scoping just `docs`:
`"dev:docs": "turbo run dev --parallel --scope 'docs'"`

But I'd like to also add other packages with the star selector which doesn't seem to work:
`"dev:docs": "turbo run dev --parallel --scope '@org-name/*,docs'"`

**Reply by jaredpalmer** [reply]
At the moment, `turbo` doesn't support spaces in CLI flags `--scope="org-name/*"` is good. `--scope "org-name/*"` is not.

**Reply by souporserious** [reply]
Oops, must have spelled that wrong, my bad! So it looks like there might be an issue with `@` org names possibly? The following works and scopes things fine:

```bash
"dev": "turbo run dev --parallel --scope='figma-plugin' --scope='docs'"
```

However, it seems to ignore our other company packages in the same repo `--scope='@company/package'` or `--scope='@company/*'`. I'll try to get a repro up next week when I'm back. They're private NPM packages, but assuming that shouldn't matter?
